python_modules/jupedsim/jupedsim/sqlite_serialization.py

```diff
@@ -35,6 +35,8 @@ def __init__(
         *,
         output_file: Path,
         every_nth_frame: int = 4,
+        buffer_in_memory: bool = False,
+        max_buffered_frames: int = 1000,
```

Comment:
With no manual buffering this probably should one additional parameter: `commit_every_nth_write: int = 100` Or some other sensible default level.